### DISCUSSION TITLE
What PNG setting is causing issues when creating a layered PSD file?

### DISCUSSION THREAD

**Discussion by vikeshshah85** [ImageMagick/ImageMagick#3369] [Help] [answered]
Hello all,

_ImageMagick (7.0.10-61 Q16 x64 2021-01-30)
Windows 10 64 bit
Adobe Photoshop 22.2.0_

I'm completely new to ImageMagick (and also image editing in general!).  I've been trying to follow [@fmw42](https://github.com/fmw42)'s instructions from this [old ImageMagick discourse community post](https://legacy.imagemagick.org/discourse-server/viewtopic.php?p=138846&sid=305bf533ed5f59f527a8355fd49518af#p138846) to create a PSD file from two images:

- [model.jpg](https://drive.google.com/file/d/1Ss23a1FVhUlb6HSsExKVKCQnApd8C6b2/view?usp=sharing)            - photo of model (5792x8668) & base layer
- [render.png](https://drive.google.com/file/d/1gW_YU3OU9KNKw_1Zs4bS6hx9ZxpsEsgk/view?usp=sharing)           - render of garment (5792x8668)

[**desired-result.psd**](https://drive.google.com/file/d/19nAeZ0P2X-0vw6xvi17ZzaHi6eKbTBQm/view?usp=sharing) 

![image](https://user-images.githubusercontent.com/25871520/110256908-433e2b80-7f93-11eb-92df-4decabf89f47.png)

This is the code I used and result:

```
magick render.png model.jpg ( -clone 1,0 -background none -flatten -alpha off ) -reverse result.psd`
```

![image](https://user-images.githubusercontent.com/25871520/110256982-b051c100-7f93-11eb-8cc0-4ceb20f78df4.png)
- [**result.psd**](https://drive.google.com/file/d/19nAeZ0P2X-0vw6xvi17ZzaHi6eKbTBQm/view?usp=sharing)

You can see in the picture that something strange happens to the garment render layer as it is enlarged, not centred and duplicated.  Does anyone know if there is a setting in the **render.png** causing this issue?  

I believe the issue is being caused by the **render.png** because I tested a few things relating to its alpha channel including creating an image with it removed and switched off.  When I combine it with **model.jpg** in the psd, a similar issue happens again.  

```
magick render.png -alpha remove -alpha off render_alpha_removeoff.png
magick render_alpha_removeoff.png model.jpg ( -clone 1,0 -background none -flatten -alpha off ) -reverse result-alpha_removeoff.psd
```

![image](https://user-images.githubusercontent.com/25871520/110258608-61a82500-7f9b-11eb-8c69-8497eb8f3e2e.png)

However, when I convert the **render_alpha_removeoff.png** to a jpg and then back to a png, it doesn't seem to suffer from the same issue:

```
magick render_alpha_removeoff.png render_s_alpha_removeoff.jpg
magick render_s_alpha_removeoff.jpg render_s_alpha_removeoff-viajpg.png
magick render_s_alpha_removeoff-viajpg.png model.jpg ( -clone 1,0 -background none -flatten -alpha off ) -reverse result-alpha_removeoff-viajpg.psd
```

![image](https://user-images.githubusercontent.com/25871520/110258861-94065200-7f9c-11eb-9193-27aede7dfde4.png)

I've also noticed that the problem does not happen when I import the PNGs into a PSD by itself...

```
magick render_alpha_removeoff.png ( -clone 0 -background none -flatten -alpha off ) result-renderonly_alpha_removeoff_pnginput.psd
```

![image](https://user-images.githubusercontent.com/25871520/110259109-f3189680-7f9d-11eb-8952-fd58efddeac7.png)

Is anyone able to help me work out what might be causing it in the PSD?

I've collated assets into a Google Drive folder in case someone is able to look at them and help:

- [**Google Drive folder with image assets and PSDs**](https://drive.google.com/drive/folders/11nciboB04zm4zcYatMW7b5-T9uPEpoT5?usp=sharing)
-- [_Documentation describing tests I've done_](https://drive.google.com/file/d/1qKxjhLibG-vqKC6ITJBIiDvZRHM7wG59/view?usp=sharing)

Thanks for any help you can provide.

```
Microsoft Windows [Version 10.0.19042.804]
(c) 2020 Microsoft Corporation. All rights reserved.

C:\Users\vikes\Desktop\Working\ImageMagick\Test Shared>magick -version
Version: ImageMagick 7.0.10-61 Q16 x64 2021-01-30 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192829336
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib

```

**Comment by fmw42** [comment]
Try adding +repage after render.png. It may have a virtual canvas with offsets.

**Reply by vikeshshah85** [reply]
Thanks @fmw42 - I really appreciate the help you've given to everyone in the community.  I wouldn't have been able to get to this stage if it wasn't for your earlier posts!

I've tried the following lines with +repage included in different positions but it still seems to suffer from the same issue.

Is there anything else I should test?

Happy to give anything a go!


```
magick render.png +repage model.jpg ( -clone 1,0 -background none -flatten -alpha off ) -reverse result-repage.psd
magick render.png model.jpg +repage ( -clone 1,0 -background none -flatten -alpha off ) -reverse result-repage2.psd
magick model.jpg render.png +repage ( -clone 0,1 -background none -flatten -alpha off ) result-repage3.psd
```

![image](https://user-images.githubusercontent.com/25871520/110260578-bc924a00-7fa4-11eb-927d-a6dad26e5bed.png)
![image](https://user-images.githubusercontent.com/25871520/110260608-e2b7ea00-7fa4-11eb-899e-1a5eb5fca084.png)
![image](https://user-images.githubusercontent.com/25871520/110260856-d3856c00-7fa5-11eb-97ab-1a96d85f97a8.png)

**Comment by fmw42** [comment, marked as answer]
The issue is that you are mixing one image at depth 8 and the other at depth 16. Make them both depth 8 by using -depth 8 before the output. So this works for me on my Mac.  If you are on Windows, remove the backslashes.

`magick render.png model.jpg \( -clone 1,0 -background none -flatten -alpha off +write tmp.png \) -reverse -depth 8 result.psd`

**Reply by vikeshshah85** [reply]
Thank you so much for explaining 8 bit vs. 16 bit depth issue @fmw42.  From reading your earlier forum posts, I suspected you might have been able to work out the solution.  I really appreciate the quick help.  Thank you for continuing to help the community and newbs such as myself!

**Comment by snibgo** [comment]
You might find that using "-depth 16" instead of "-depth 8" gives a 16-bit PSD, which might suit your purposes better. I don't use Photoshop and can't test this.
